src/py/flwr/server/superlink/state/in_memory_state.py

```diff
@@ -91,11 +91,13 @@ def get_task_ins(
                     and task_ins.task.consumer.anonymous is False
                     and task_ins.task.consumer.node_id == node_id
                     and task_ins.task.delivered_at == ""
+                    and task_ins.task.created_at + task_ins.task.ttl > time.time()
                 ) or (
                     node_id is None  # Anonymous
                     and task_ins.task.consumer.anonymous is True
                     and task_ins.task.consumer.node_id == 0
                     and task_ins.task.delivered_at == ""
+                    and task_ins.task.created_at + task_ins.task.ttl > time.time()
```

Comment:
@mohammadnaseri nit: it might be more reliable to get the time once (before the if statement) and then use the same timestamp in both comparisons